<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="rolename" default="hero" />
	<arg name="host" default="localhost" />
	<arg name="port" default="2000" />
	<arg name="max_speed" default="7.5" />
	<arg name="opp_speed" default="8.0" />
	<arg name="plan_horizon" default="3.0" />
	<arg name="plan_steps" default="10" />

	<!-- bridge -->
	<include file="$(find carla_ros_bridge)/launch/carla_ros_bridge.launch">
		<arg name="host" value="$(arg host)" />
		<arg name="port" value="$(arg port)" />
		<arg name="fixed_delta_seconds" value="false" />
	</include>

	<!-- shared by all the cars  -->
	<!-- <node name="car_filter" pkg="carla_circle" type="car_filter.py" output="screen" /> -->

	<include file="$(find carla_circle)/launch/car_simulation.launch">
		<arg name="rolename" value="$(arg rolename)" />
		<arg name="color" value="255, 0, 0" />
		<arg name="spawn_point" default="46.8, 15.0, 2.0, 0.0, 0.0, 90.0" />
		<arg name="max_speed" value="$(arg max_speed)" />
		<arg name="opp_speed" value="$(arg opp_speed)" />
		<arg name="plan_horizon" value="$(arg plan_horizon)" />
		<arg name="plan_steps" value="$(arg plan_steps)" />
	</include>

	<!-- visualization -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find carla_circle)/resource/carla_default_rviz.cfg.rviz" />

</launch>
